### ISSUE TITLE
Problem in 8 alpha 1 with user_webdavauth and "User" app options

### ISSUE THREAD

**Issue by canepan** [owncloud/core#13080]
I installed the new alpha 1, executed the wizard and enabled  the user_webdavauth app.
Now, when changing options in the “Users” app (as an admin), I am unable to make the settings persistent, except for quota (which is saved)

**Comment by karlitschek** [comment]
@MorrisJobke @blizzz @LukasReschke Can you have a look please?

**Comment by LukasReschke** [comment]
Well - that's to my knowledge a bug in user_webdavauth which persist since years. Same reason why you can't add users etc…

My https://github.com/owncloud/core/pull/12799 would have fixed this and a lot of other related bugs. But since we're now in feature freeze and this is quite a big change I'm afraid to say that fixing this should wait for 8.1.

**Comment by LukasReschke** [comment]
We have a lot of other similar issues open but since I'm on my iPhone searching issues is kinda pain - I'm fine with keeping this open since it is now linked to my PR.

**Comment by LukasReschke** [comment]
I think I also referenced this issue in a PR of the issue opener and asked for testing which unfortunately nobody did.

**Comment by RealRancor** [comment]
>We have a lot of other similar issues open but since I'm on my iPhone searching issues is kinda pain - I'm fine with keeping this open since it is now linked to my PR.

Probably: https://github.com/owncloud/core/issues/8749 and the linked issue there.

**Comment by MorrisJobke** [comment]
@LukasReschke Did you want to reference #12799?

**Comment by LukasReschke** [comment]
Yes.

**Comment by MorrisJobke** [comment]
So this issue will be postponed to 8.1?

**Comment by LukasReschke** [comment]
Well - it existed since ever to my understanding and I really don't want to merge that PR now.

**Comment by blizzz** [comment]
I second @LukasReschke for postponing: a changeset too big for including it post freeze.

**Comment by karlitschek** [comment]
agreed

**Comment by LukasReschke** [comment]
https://github.com/owncloud/core/pull/12799 is updated and ready for review. Let's see if nowadays somebody bothers to review it – otherwise it will simply not make it for 8.1 as well ;-)

@canepan Please test. THX!

**Comment by canepan** [comment]
I tried downloading the 8.1 version from here: 
https://github.com/owncloud/core/tree/d25340cd171cee9487132ea1bcf5015b8e56bb09, 
installing it alongside the 7.0.4 I have running now, but accessing the 
URL (without config, so I was expecting the wizard) I receive an error:
> *Fatal error*: Call to a member function getSession() on a non-object 
> in 
> */rnd/apps/servers/httpd-2.4/htdocs/owncloud81/lib/private/template.php* 
> on line *53*
Am I missing something?
I use PHP 5.5

Nicola

Il 12/02/15 15:11, Lukas Reschke ha scritto:

**Comment by LukasReschke** [comment]
It will only be possible with the ownCloud master branch :-)

You possibly have to download this and the 3rdparty repo from GitHub or use a nightly build. Though I don't know a URL for the daily builds?

@karlitschek Can you help here maybe?

**Comment by canepan** [comment]
By downloading the 3rdparty (and adding apps, gallery, files_textviewer) 
from github, I was able to make the wizard work.

Thank you for the pointer.

Nicola

Il 13/02/15 09:48, Lukas Reschke ha scritto:

**Comment by canepan** [comment]
After installing OwnCloud 8.1 (from github master branch) and the 
required module (3rdparty), enabling thee webdav backend, I found the 
following problems:
- when logging in with a webdav user, and loggin back in as admin, I 
cannot see the user to assign him a group.
- I'm unable to save display settings in "users" app (i.e. user backend, 
last login, storage location)

Nicola

Il 13/02/15 09:48, Lukas Reschke ha scritto:

**Comment by LukasReschke** [comment]
Are you on the correct branch of the application?

**Comment by canepan** [comment]
How can I check it?
I got the core from here: https://github.com/owncloud/core/tree/d25340cd171cee9487132ea1bcf5015b8e56bb09
And then apps, 3rdparty and other apps from the master branch of their repos.

**Comment by karlitschek** [comment]
Can you please test with the latest 8.1 alpha3 ?
Thanks

**Comment by canepan** [comment]
I did the following:
- unpack the archive
- create data directory and give web server user write access to it
- followed the web wizard
- enabled webdav_userauth
- disabled unused plugins (activity, updater, first run wizard)
- applied my theme and enabled following options in config.php:
>   'theme' => 'XXX',
>   'appstoreenabled' => false,
>   'updatechecker' => false,
>   'has_internet_connection' => false,
>   'log_authfailip' => true,
>   'maxZipInputSize' => 1073741824,
>   'allowZipDownload' => true,
>   'loglevel' => '2',
>   'session_lifetime' => 3600,
>   'forcessl' => true,
>   'user_webdavauth_url' => 'MY_WEBDAV_URL',
>   'maintenance' => false,
>   'remember_login_cookie_lifetime' => 86400,
>   'check_for_working_webdav' => false,
- open User config
- enable " Show storage location", " Show last log in" and "Show user 
backend", set quota to 1GB
- go to files
- go back to users: only quota option was kept
- user creation is not possible (error is "user already exists"): this 
is correct (but error could be more specific)
- login as a webdav user: OK
- login as an admin: user that logged in was not present (so I could not 
add it to any group)

Nicola

Il 04/05/15 19:32, Frank Karlitschek ha scritto:

**Comment by MorrisJobke** [comment]
> - go back to users: only quota option was kept

Duplicate of #12727

> - user creation is not possible (error is "user already exists"): this is correct (but error could be more specific)

What should it be?

> - login as an admin: user that logged in was not present (so I could not add it to any group)

So you couldn't log in as admin, or what is the problem here?

**Comment by canepan** [comment]
Sorry, I try to be more precise (see below)

Il 05/05/15 11:32, Morris Jobke ha scritto:
>
>       * go back to users: only quota option was kept
>
> Duplicate of #12727 <https://github.com/owncloud/core/issues/12727>
>
>       * user creation is not possible (error is "user already
>         exists"): this is correct (but error could be more specific)
>
> What should it be?
>
Something like "backend does not allow to create users"
>
>       * login as an admin: user that logged in was not present (so I
>         could not add it to any group)
>
> So you couldn't log in as admin, or what is the problem here?
>
I logged in as admin, but the webdav user (let's call it "nicola") does 
not appear in users list, so I'm unable to assign groups to him

**Comment by MorrisJobke** [comment]
> I logged in as admin, but the webdav user (let's call it "nicola") does not appear in users list, so I'm unable to assign groups to him

@LukasReschke Is this even possible with the webdav user backend? Wasn't there a restriction?

> Something like "backend does not allow to create users"

Got it.

**Comment by MorrisJobke** [comment]
Maybe this will be solved once #12799 is merged into master?

**Comment by DeepDiver1975** [comment]
Solved with #19229
